barbarik.py

```diff
@@ -171,10 +171,7 @@ def getSolutionFromUniGen(inputFile, numSolutions, indVarList, newSeed):
                 freq = int(line.split(':')[-1])
                 for i in range(freq):
                     solList.append(line.split(':')[0].replace('v', '').strip())
-                    if (len(solList) == numSolutions):
-                        break
-                if (len(solList) == numSolutions):
-                    break
```

Comment:
This seems wrong. We should be returning `numSolutions` here only.  Why do we need to change this?

Reply:
Let's say we ask the sampler for 10 uniformly random solutions and it returns 20. The current implementation will only use the first 10 samples and discard the remaining 10. This is OK if the samples were returned in a randomised order. However, this is not ok if the sampler returned the samples in some sorted order.

The right thing to do in this case would be to randomly sample 10 out of the 20. This is exactly what we do in the very next line (178-180). 

AppMC3 and Unigen do actually return the samples in some sorted order, and thus cause the test to Reject incorrectly.